R/transactions.R

```diff
@@ -43,4 +57,29 @@ setGeneric("dbCommit",
 setGeneric("dbRollback",
   def = function(conn, ...) standardGeneric("dbRollback"),
   valueClass = "logical"
-)
\ No newline at end of file
+)
+
+#' @aliases withTransaction,DBIConnection-method
+#' @export
+#' @rdname transactions
+setGeneric("withTransaction",
+  def = function(conn, statement, ...) standardGeneric("withTransaction"),
+  valueClass = "logical"
+)
+
+#' @export
+setMethod("withTransaction", "DBIConnection", function(conn, statement, ...) {
```

Comment:
Just to make this explicit:

- this version currently supports any SQL statement to be passed on to `withTransaction`:
```r
withTransaction(con, 
  "INSERT INTO cars (speed, dist) VALUES (1, 1), (2, 2), (3, 3);")
```

- @hadley would like to see this changed to:
```r
f <- function() {
  dbExecQuery(con, 
    "INSERT INTO cars (speed, dist) VALUES (1, 1), (2, 2), (3, 3);")
}
withTransaction(con, f)
```
Or is it this:
```r
f <- function() {
  "INSERT INTO cars (speed, dist) VALUES (1, 1), (2, 2), (3, 3);"
}
withTransaction(con, f)
```

- @krlmlr would like to see this changed to:
```r
withTransaction(con, dbExecQuery(con, 
  "INSERT INTO cars (speed, dist) VALUES (1, 1), (2, 2), (3, 3);"))
```

Did I get something wrong there? In either case, what would be the advantages vis-a-vis what it supports right now (simple SQL statement)?